client/state/sites/plans/test/selectors.js

```diff
@@ -53,6 +55,73 @@ describe( 'selectors', () => {
 			expect( plans ).to.eql( plans1 );
 		} );
 	} );
+	describe( '#getCurrentPlan()', () => {
+		context( 'when no plan data is found for the given siteId', () => {
+			const siteId = 77203074;
+			const state = deepFreeze( {
+				sites: {
+					plans: {
+						[ siteId ]: {}
+					}
+				}
+			} );
+
+			it( 'returns null', () => {
+				const plan = getCurrentPlan( state, siteId );
+				expect( plan ).to.eql( null );
+			} );
+		} );
+
+		context( 'when plans are found for the given siteId', () => {
+			context( 'when those plans include a \'currentPlan\'', () => {
+				const siteId = 77203074;
+				const plan1 = { currentPlan: true };
+				const state = deepFreeze( {
+					sites: {
+						plans: {
+							[ siteId ]: {
+								data: [ plan1 ]
+							}
+						},
+						items: {
+							[ siteId ]: {
+								URL: 'https://example.wordpress.com'
+							}
+						}
+					}
+				} );
+
+				it( 'returns the currentPlan', () => {
+					const plan = getCurrentPlan( state, siteId );
+					expect( plan ).to.eql( plan1 );
+				} );
+			} );
+
+			context( 'when those plans do not include a \'currentPlan\'', () => {
+				const siteId = 77203074;
+				const plan1 = { currentPlan: false };
+				const state = deepFreeze( {
+					sites: {
+						plans: {
+							[ siteId ]: {
+								data: [ plan1 ]
+							}
+						},
+						items: {
+							[ siteId ]: {
+								URL: 'https://example.wordpress.com'
+							}
+						}
+					}
+				} );
+
+				it( 'returns a new sitePlanObject', () => {
+					const plan = getCurrentPlan( state, siteId );
+					expect( plan ).to.eql( {} );
```

Comment:
Thanks we'll probably clean this one up when we remove that fallback.

For anyone reading this is coming from the following in plans/assembler.js
```javascript
const createSitePlanObject = ( plan ) => {
	if ( ! plan ) {
		return {};
	}

```